mutiny-core/src/labels.rs

```diff
@@ -131,6 +146,16 @@ pub trait LabelStorage {
     fn edit_contact(&self, id: impl AsRef<str>, contact: Contact) -> Result<(), MutinyError>;
     /// Gets all the existing tags (labels and contacts)
     fn get_tag_items(&self) -> Result<Vec<TagItem>, MutinyError>;
+    /// Finds a contact that has the given lnurl as either a lnurl or a lightning address
+    fn get_contact_for_lnurl(&self, lnurl: &LnUrl) -> Result<Option<String>, MutinyError> {
+        let contacts = self.get_contacts()?;
+        for (id, contact) in contacts {
```

Comment:
And if there's hundreds of contacts? That feels pretty inefficient. Probably needs a lookup or something.

Reply:
We'd have to change how we store contacts. The payment will always be the longest part of this so figured the slight in efficiency here isn't the end of the world